tests/api/v1/datadog/api_logs_test.go

```diff
@@ -73,10 +73,17 @@ func TestLogsList(t *testing.T) {
 	// Find first log item
 	logsRequest.SetLimit(1)
 
-	logsResponse, httpresp, err = Client(ctx).LogsApi.ListLogs(ctx).Body(logsRequest).Execute()
+	err = tests.Retry(time.Duration(5)*time.Second, 30, func() bool {
+		logsResponse, httpresp, err = Client(ctx).LogsApi.ListLogs(ctx).Body(logsRequest).Execute()
+		if err != nil {
+			t.Fatalf("Error listing logs: Response %s: %v", err.(datadog.GenericOpenAPIError).Body(), err)
```

Comment:
Isn't using `Fatalf` inside the retry still breaking the test at the first failure, and rendering the retry useless ?

Reply:
The retry here still works because it is making sure the length of the response was what we expected as thats where the main failures were coming from. But yeah we should be allowing for non 2xx responses here too.